### PR TITLE
Snp 612

### PR DESCRIPTION
**Pull request by maskimko**
Able to display past task outputs form S3

### CODE REVIEW

**Review by remm — APPROVED** [review]

File: storer/s3helpers.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package storer
```
Comment (matancwix) [inline storer/s3helpers.go:1]: I would call it s3handler and not s3helper. It's more common in the code examples that I saw.

File: storer/s3.go [hunk 2]
```diff
@@ -72,19 +67,33 @@ func S3UploadTaskWithSuffix(bucket string, id int, suffix *string) error {
 	return nil
 }
 
-func getCredentialsProvider() (credentials.Provider, error) {
+func S3DownloadTask(bucket string, id int) error {
+	return S3DownloadTaskWithSuffix(bucket, id, nil)
+}
+
+func S3DownloadTaskWithSuffix(bucket string, id int, suffix *string) error {
```
Comment (matancwix) [inline storer/s3.go:74]: looks much better now 👍

File: launcher/emitter.go [hunk 3]
```diff
@@ -95,3 +115,38 @@ func GetCompletedTaskOutput(taskId int) ([]string, error) {
 	}
 	return lines, nil
 }
+
+func PullS3TaskOutput(taskId int) error {
+	tm := GetTaskManager()
+	task := tm.Get(taskId)
+	if task == nil {
```
Comment (matancwix) [inline launcher/emitter.go:122]: There is a "Christmas tree" problem with this code (if inside if....). Please slack me to elaborate.

**Comment by matancwix** [comment]
We spoke about most of the things on the code review. so I didn't write them as comments.

**Pull request merged**